### PR TITLE
Joseph Bravo / completed: array-manipulation-2

### PR DESCRIPTION
**Pull request by joseph-bravo**
image

### CODE REVIEW

File: array-manipulation-2/src/take-right.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+output an array
+loop through array while counter is less than count arg AND array.length - i - 1 doesn't go below 0
+  start loop index from the end, then decrement at end of loop
+  unshift array@index
+return output
+ */
+
+function takeRight(array, count) {
+  var output = [];
+  for (var i = 0; i < count && (array.length - i - 1 >= 0); i++) {
```
Comment (codingcodymiller) [inline array-manipulation-2/src/take-right.js:12]: This is pretty complicated. I would just start the loop at `array.length - count`, that can help you simplify this by a lot.

**Review by codingcodymiller — CHANGES_REQUESTED** [review]

File: array-manipulation-2/src/take-right.js [hunk 2]
```diff
@@ -1,17 +1,17 @@
...
 
 function takeRight(array, count) {
   var output = [];
-  for (var i = 0; i < count && (array.length - i - 1 >= 0); i++) {
-    var index = array.length - i - 1;
-    output.unshift(array[index]);
+  for (var i = array.length - count; i < array.length; i++) {
+    if (i >= 0) {
+      output.push(array[i]);
+    }
```
Comment (joseph-bravo) [inline array-manipulation-2/src/take-right.js:14]: !

File: array-manipulation-2/src/take-right.js [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+
+/*
+loop through array starting at given length - index,
+  check if i is within range of indexes (greater than or equal 0)
+push into output array then return that
+*/
+
+function takeRight(array, count) {
+  var output = [];
+  for (var i = array.length - count; i < array.length; i++) {
```
Comment (codingcodymiller) [inline array-manipulation-2/src/take-right.js:11]: How about `var i = Math.max(0, array.length - count)`?

If you do this, you can remove the need to have a conditional within the loop

**Review by codingcodymiller — CHANGES_REQUESTED** [review]

**Comment by joseph-bravo** [comment]
image

**Review by codingcodymiller — APPROVED** [review]

**Pull request merged**